### PR TITLE
feat(moddingway): add support for editing strikes

### LINKED ISSUE #146: Edit a strike
As a moderator, I want to be able to edit an existing strike so that I can correct or update information as needed.

- Acceptance Criteria:
  - Command should be /editstrike
  - The moderator can specify the Strike ID to identify which note to edit.
  - The bot should allow the moderator to update the Strike Reason field (limit of 300 chars).
  - The bot should allow the moderator to update the Strike Severity. This field should be optional. The points should also be updated.
  - If a strike is edited, the bot should DM the edit to the user.
  - The bot will confirm that the strike has been successfully updated.
  - The bot will replace the ID of the moderator in the original record with the editor's.

### PR DESCRIPTION
**Pull request by a1icja**
## Description

Adds support for editing strikes incl. automatically updating points against the user. Also updates to uv 0.11.22 to be in-line with what brew has on macOS.

### Related Ticket

Closes #146 

## Type of Change

New feature

## Testing

Tested in NAUR testing server. Results can be viewed in [this channel](https://canary.discord.com/channels/1293062334566109206/1517595604736413696)

## Checklist

- [X] Self-reviewed the code
- [ ] Updated documentation (if needed)
  - Not aware of any at the moment
- [X] Tests added and passing (if applicable)
- [X] Needs QA

---

> [!IMPORTANT]
> Checking **Needs QA** triggers an automated checklist.
> Open in **Draft mode** so you can fill it out before requesting review.

### CODE REVIEW

File: services/moddingway/moddingway/services/strike_service.py [hunk 1]
```diff
@@ -134,6 +135,112 @@ async def delete_strike(logging_embed: discord.Embed, strike_id: int) -> str:
...
+    if strike is None:
+        log_info_and_add_field(
+            logging_embed,
+            logger,
+            "Result",
+            "Strike not found",
+        )
+        errors.append("Strike not found in database")
+
+    if reason is None and severity is None:
```
Comment (brtran4) [inline services/moddingway/moddingway/services/strike_service.py:159]: looks like this should be an `or`

File: services/moddingway/moddingway/services/strike_service.py [hunk 2]
```diff
@@ -134,6 +135,112 @@ async def delete_strike(logging_embed: discord.Embed, strike_id: int) -> str:
...
+            temporary_points_to_remove = _get_severity_points(old_severity)
+
+        users_database.decrement_user_strike_points(
+            strike.user_id, temporary_points_to_remove, permanent_points_to_remove
+        )
+
+        # refetch after update
+        db_user = users_database.get_user_by_internal_id(strike.user_id)
+        if db_user is None:
+            return "User not found in database"
```
Comment (brtran4) [inline services/moddingway/moddingway/services/strike_service.py:197]: do we need to refetch when we already checked if the user exists above?

File: services/moddingway/moddingway/services/strike_service.py [hunk 3]
```diff
@@ -134,6 +135,112 @@ async def delete_strike(logging_embed: discord.Embed, strike_id: int) -> str:
...
+    )
+
+    strike_user = await client.fetch_user(int(strike.discord_user_id))
+    await send_dm(
+        logging_embed,
+        strike_user,
+        "Your strike in NA Ultimate Raiding - FFXIV has been updated.\n"
+        f"**Reason:** {new_reason}\n**Severity:** {new_severity.name}",
+        context="Strike edit",
+    )
```
Comment (brtran4) [inline services/moddingway/moddingway/services/strike_service.py:240]: let's ask nova and gogo if they want a dm sent and if this text is good

**Review by brtran4 — CHANGES_REQUESTED** [review]

File: services/moddingway/moddingway/services/strike_service.py [hunk 4]
```diff
@@ -134,6 +135,112 @@ async def delete_strike(logging_embed: discord.Embed, strike_id: int) -> str:
...
+    if reason is None and severity is None:
+        errors.append("No updates provided. Please specify a reason and/or severity.")
+
+    if reason is not None and len(reason) > MAX_STRIKE_REASON_LENGTH:
+        errors.append(
+            f"Reason is too long (max {MAX_STRIKE_REASON_LENGTH} characters). "
+            "Please shorten the strike reason."
+        )
+
+    # first condition won't happen, but unresolved-attribute still complains
```
Comment (brtran4) [inline services/moddingway/moddingway/services/strike_service.py]: should be able to resolve with `# ty: ignore[unresolved-attribute]`

File: services/moddingway/moddingway/services/strike_service.py (hunk 2 above)
Comment (a1icja) [reply services/moddingway/moddingway/services/strike_service.py:197]: We're refetching the object after the `decrement_user_strike_points` is applied

File: services/moddingway/moddingway/services/strike_service.py (hunk 1 above)
Comment (a1icja) [reply services/moddingway/moddingway/services/strike_service.py:159]: It's an `and` because you can do the following:
- reason only
- severity only
- reason and severity

Since these values were already set when the strike was originally created, we can fallback on existing values if we don't want to update them

File: services/moddingway/moddingway/services/strike_service.py (hunk 3 above)
Comment (a1icja) [reply services/moddingway/moddingway/services/strike_service.py:240]: Confirmed with them it's okay

File: services/moddingway/moddingway/services/strike_service.py (hunk 4 above)
Comment (a1icja) [reply services/moddingway/moddingway/services/strike_service.py]: Requires quite a few of those but added

File: services/moddingway/moddingway/services/strike_service.py (hunk 1 above)
Comment (brtran4) [reply services/moddingway/moddingway/services/strike_service.py:159]: ok yeah that sounds good

**Review by brtran4 — APPROVED** [review]

**Review by MKShadowZX — COMMENTED** [review]
Was able to successfully perform each tests in the list, including the edge cases, and made sure they work accordingly. The only thing though, was that I added libc6-dev \  to the dockerfile after "gcc \ " and it worked after. Running make run as is gave an error which I posted the full log in a .txt and uploaded it in the NAUR moddingway channel.

**Review by MKShadowZX — APPROVED** [review]
